ipatests/test_integration/test_adtrust_install.py

```diff
@@ -0,0 +1,190 @@
+#
+# Copyright (C) 2020  FreeIPA Contributors see COPYING for license
+#
+
+"""This module provides tests for ipa-adtrust-install utility"""
+
+import re
+import textwrap
+
+from ipaplatform.paths import paths
+from ipapython.dn import DN
+from ipatests.pytest_ipa.integration import tasks
+from ipatests.test_integration.base import IntegrationTest
+
+
+class TestIpaAdTrustInstall(IntegrationTest):
+    topology = 'line'
+    num_replicas = 1
+
+    def unconfigure_replica_as_agent(self, host):
+        """ Remove a replica from the list of agents.
+
+        cn=adtrust agents,cn=sysaccounts,cn=etc,$BASEDN contains a list
+        of members representing the agents. Remove the replica principal
+        from this list.
+        """
+        remove_agent_ldif = textwrap.dedent("""
+             dn: cn=adtrust agents,cn=sysaccounts,cn=etc,{base_dn}
+             changetype: modify
+             delete: member
+             member: fqdn={hostname},cn=computers,cn=accounts,{base_dn}
+             """.format(base_dn=host.domain.basedn, hostname=host.hostname))
+        # ok_returncode =16 if the attribute is not present
+        tasks.ldapmodify_dm(self.master, remove_agent_ldif,
+                            ok_returncode=[0, 16])
+
+    def test_samba_config_file(self):
+        """Check that ipa-adtrust-install generates sane smb.conf
+        This is regression test for issue
+        https://pagure.io/freeipa/issue/6951
+        """
+        self.master.run_command(
+            ['ipa-adtrust-install', '-a', self.master.config.admin_password,
+             '--add-sids', '-U'])
+        res = self.master.run_command(['testparm', '-s'])
+        assert 'ERROR' not in (res.stdout_text + res.stderr_text)
+
+    def test_add_agent_not_allowed(self):
+        """Check that add-agents can be run only by Admins
+        """
+        user = "nonadmin"
+        passwd = "Secret123"
+        tasks.create_active_user(self.master, user, passwd,
```

Comment:
Created user should be removed in teardown.
Also could you please explain why it is needed? Does `curl` somehow use the kerberos ticket? If yes, please add the comment as it is hacky and non-obvious

Reply:
OK, added a comment and also try/finally block.